src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -869,6 +870,17 @@ object FlexibleDataModelRelationUtils {
         s"""Error parsing value of field '$propertyName' as an array of timestamps: ${e.getMessage}""".stripMargin)
     }
 
+  private def extractSpaceOrDefault(schema: StructType, row: Row, defaultSpace: Option[String]) =
+    defaultSpace.map(Right(_)).getOrElse(extractSpace(schema, row)).leftMap { e =>
+      new CdfSparkIllegalArgumentException(
```

Comment:
can you add a test case to make sure we capture this Exception when the `instanceSpace` is not provided? 😁